In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../data/raw/mec02_data_centres.csv")
df

,Statistic Label,Quarter,Electricity Consumption,UNIT,VALUE
0,Gigawatt hours,2015Q1,All metered electricity consumption,GWh,6565
1,Gigawatt hours,2015Q1,Data centres,GWh,291
2,Gigawatt hours,2015Q1,Customers other than data centres,GWh,6274
3,Gigawatt hours,2015Q2,All metered electricity consumption,GWh,6086
4,Gigawatt hours,2015Q2,Data centres,GWh,304
...,...,...,...,...,...
127,Gigawatt hours,2025Q3,Data centres,GWh,1956
128,Gigawatt hours,2025Q3,Customers other than data centres,GWh,5778
129,Gigawatt hours,2025Q4,All metered electricity consumption,GWh,8420
130,Gigawatt hours,2025Q4,Data centres,GWh,1991


In [3]:
wide = df.pivot(index="Quarter", columns = "Electricity Consumption", values= "VALUE",)
wide

Electricity Consumption,All metered electricity consumption,Customers other than data centres,Data centres
Quarter,,,
2015Q1,6565,6274,291
2015Q2,6086,5783,304
2015Q3,5783,5467,316
2015Q4,6165,5836,329
2016Q1,6651,6310,341
2016Q2,6471,6111,361
2016Q3,5926,5540,386
2016Q4,6307,5912,395
2017Q1,6867,6460,407


In [4]:
wide = wide.rename(columns={
    "All metered electricity consumption": "total",
    "Customers other than data centres": "others",
    "Data centres": "data_centres"
})
wide

Electricity Consumption,total,others,data_centres
Quarter,,,
2015Q1,6565,6274,291
2015Q2,6086,5783,304
2015Q3,5783,5467,316
2015Q4,6165,5836,329
2016Q1,6651,6310,341
2016Q2,6471,6111,361
2016Q3,5926,5540,386
2016Q4,6307,5912,395
2017Q1,6867,6460,407


In [5]:
wide["dc_share"] = wide["data_centres"] / wide["total"] * 100
wide.round(1)

Electricity Consumption,total,others,data_centres,dc_share
Quarter,,,,
2015Q1,6565,6274,291,4.4
2015Q2,6086,5783,304,5.0
2015Q3,5783,5467,316,5.5
2015Q4,6165,5836,329,5.3
2016Q1,6651,6310,341,5.1
2016Q2,6471,6111,361,5.6
2016Q3,5926,5540,386,6.5
2016Q4,6307,5912,395,6.3
2017Q1,6867,6460,407,5.9


In [6]:
wide["year"] = wide.index.str[:4]
yearly = wide.groupby("year")[["data_centres", "others", "total"]].sum()
yearly["dc_share"] = yearly["data_centres"] / yearly["total"] * 100
yearly.round(1)

Electricity Consumption,data_centres,others,total,dc_share
year,,,,
2015,1240,23360,24599,5.0
2016,1483,23873,25355,5.8
2017,1762,23963,25726,6.8
2018,2183,24548,26730,8.2
2019,2489,24016,26504,9.4
2020,3031,24026,27056,11.2
2021,4013,24494,28506,14.1
2022,5274,24552,29825,17.7
2023,6339,24242,30581,20.7


how fast did each group grow each year??

In [7]:

yearly["dc_growth"] = yearly["data_centres"].pct_change() * 100
yearly["others_growth"] = yearly["others"].pct_change() * 100
yearly.round(1)

Electricity Consumption,data_centres,others,total,dc_share,dc_growth,others_growth
year,,,,,,
2015,1240,23360,24599,5.0,NaN,NaN
2016,1483,23873,25355,5.8,19.6,2.2
2017,1762,23963,25726,6.8,18.8,0.4
2018,2183,24548,26730,8.2,23.9,2.4
2019,2489,24016,26504,9.4,14.0,-2.2
2020,3031,24026,27056,11.2,21.8,0.0
2021,4013,24494,28506,14.1,32.4,1.9
2022,5274,24552,29825,17.7,31.4,0.2
2023,6339,24242,30581,20.7,20.2,-1.3


2015 year shows it as NaN , because we dont have the 2014 data to compare the growth 

In 2025, share 23.2% - CSO says 23% 
         data centre growth 9.9% - CSO says 10 %
         others growth 1.6% - CSO says 2% 

Clearly , data centre demand grew upto 6 times faster than everyone else !! 

IS it speeding up or slowing down ??

In 2021, the regulator restricted new data centre connections in Dublin

By reading the dc_growth, we can see :

 2016 to 2020 - the growth is fast!, 
 2021 and 2022 - PEAK [Fastest growth in whole dataset],
 2023 - Drops to 20%, 
 2024 and 2025 - About 10% a year. 





### Growth is slowing down, and the slowdown starts right after the 2021 Dublin restrictions

### My honest takings here :

### 1. The timing is never the proof - 
slowdown matches the restrictions happened in 2021 , but it could also explain other things that happened in other years too ! 
### Other things could also explain the slowdown 
### The timing matches, but this data alone can't prove whether the restrictions caused the slowdown! 

### 2. Slower does not mean smaller - 
Growth slowed from 32% to 10%, but data centres still added about 690 GWh in 2025, and their share keeps rising every single year.
### The problem is growing more slowly, not going away.